client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,472 @@
+            </tr>
+          </thead>
+          <tbody>
+            {contest.choices.map(choice => (
+              <tr key={choice.id}>
+                <td>{choice.name}</td>
+                <td>
+                  {isTotalsSheet ? (
+                    <span>
+                      {sum(sheets.map(sheet => sheet.results[choice.id]))}
+                    </span>
+                  ) : (
+                    <>
+                      <input
+                        aria-label={`${choice.name} Votes`}
+                        className={classnames(
+                          Classes.INPUT,
+                          errors.results?.[choice.id] && Classes.INTENT_DANGER
+                        )}
+                        name={`results[${choice.id}]`}
+                        readOnly={isSubmitting}
+                        ref={register({
+                          min: 0,
+                          required: true,
+                          valueAsNumber: true,
+                        })}
+                        type="number"
+                        // Visually hide this input instead of completely unmounting it since
+                        // rendering no input at all interferes with react-hook-form's state
+                        // management, specifically its isDirty tracking
+                        style={{ display: !isEditing ? 'none' : undefined }}
+                      />
+                      {!isEditing && (
+                        <span>{selectedSheet?.results[choice.id] || 0}</span>
+                      )}
+                    </>
+                  )}
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </BatchResultTallySheetTable>
+
+        <BatchResultTallySheetButtonRow>
+          {(() => {
+            if (selectedTabId === VOTE_TOTALS_TAB_ID) {
+              return <span>Select a sheet to make edits.</span>
+            }
+            if (isEditing) {
+              return (
+                <>
+                  <Button
+                    disabled={isSubmitting}
+                    icon="delete"
+                    intent="danger"
+                    minimal
+                    onClick={discardChanges}
+                  >
+                    Discard Changes
+                  </Button>
+                  <Button
+                    icon="tick"
+                    intent="primary"
+                    loading={isSubmitting}
+                    onClick={handleSubmit(onValidSubmit)}
+                  >
+                    {sheets.length === 1 ? 'Save Tallies' : 'Save Sheet'}
+                  </Button>
+                </>
+              )
+            }
+            return (
+              <ButtonGroup>
+                <Button
+                  disabled={areResultsFinalized}
+                  icon="edit"
+                  onClick={() => setIsEditing(true)}
+                >
+                  {sheets.length === 1 ? 'Edit Tallies' : 'Edit Sheet'}
```

Comment:
On edit, can we auto focus the first tally input?

Reply:
Handled this as well as auto-focusing name for newly created sheets in https://github.com/votingworks/arlo/pull/1661/commits/744c02c4c289e7879648e2e3233dd70e41c654ad!

The latest version of react-hook-form (v7) actually provides a [`setFocus` function](https://react-hook-form.com/api/useform/setfocus), but we're on a version that doesn't include this function (v6). Either way, `autoFocus` worked well (which is React's custom polyfill of the underlying HTML `autofocus` that applies on React component mount - [reference](https://blog.maisie.ink/react-ref-autofocus/#how-does-the-autofocus-prop-work))

Reply:
Great!